### PR TITLE
feat[added item-pocessing-state]

### PR DESCRIPTION
**Pull request by Radheshyam1234**
Added the item processing state Feature. If user clicks on add to wishlist button then till the response comes from backend item will be in processing state so a spinner will be there or if a user clicks on add to cart then till the response comes from backend the add to cart btn will be disabled so that user will not be able to add same product many times in cart or wishlist on multiple clicking quickly.
wishlist spinner

### CODE REVIEW

File: src/Components/LikeBtn/LikeBtn.js [hunk 1]
```diff
@@ -10,20 +17,44 @@ export const LikeBtn = ({ prod, setToastMsg }) => {
...
         <div
           onClick={() => {
-            addProductInWishlist({ prod, setToastMsg });
+            isUserLoggedIn
+              ? addProductInWishlist({
+                  prod,
+                  setProcessingItem,
+                  setToastMsg,
+                })
+              : navigate("/login");
```
Comment (pratik-kumar-sahu) [inline src/Components/LikeBtn/LikeBtn.js:57]: - Good use of dispatch functions

File: src/Components/utils/useUserActions.js [hunk 2]
```diff
@@ -30,13 +35,15 @@ export const useUserActions = () => {
             type: "SET_WISHLIST",
             payload: response.data.response,
```
Comment (pratik-kumar-sahu) [inline src/Components/utils/useUserActions.js:36]: - You can use variable instead of hard coded string `SET_WISHLIST` for action types

**Review by pratik-kumar-sahu — APPROVED** [review]
- Good use of destructuring and react-hooks.
- Ui and Code looks clean. Go ahead with the merging

**Comment by apoorvshrimali** [comment]
Design Looks Really cool , try adding transition on hover state which will make  animation more soothing and pleasant.

**Review by Megha-Pathak — APPROVED** [review]
This is a good UX addition. Since you have created the custom hook for this now, maybe integrate this to the Move to Wishlist button too.

**Pull request merged**